crypto/x509/v3_akid.c

```diff
@@ -127,21 +127,39 @@ static AUTHORITY_KEYID *v2i_AUTHORITY_KEYID(X509V3_EXT_METHOD *method,
     }
 
     if (keyid != 0) {
-        /* prefer any pre-existing subject key identifier of the issuer cert */
+        int same_issuer = ctx->subject_cert == ctx->issuer_cert;
+        int ss = same_issuer; /* default used if ctx->issuer_pkey is not set */
+
+        ERR_set_mark();
+        if (ctx->issuer_pkey != NULL)
+            ss = X509_check_private_key(ctx->subject_cert, ctx->issuer_pkey);
+        ERR_pop_to_mark();
```

Comment:
And this block looks exactly like `cert_matches_key` to me

Reply:
Yes, but `cert_matches_key()` is defined just for apps (in `apps/lib/apps.c`).

Reply:
Ah! Sorry, my fault